File: app/src/main/java/org/schulcloud/mobile/data/model/Device.java
package org.schulcloud.mobile.data.model;

import io.realm.RealmModel;
import io.realm.annotations.PrimaryKey;
import io.realm.annotations.RealmClass;

@RealmClass
public class Device implements RealmModel {

    @PrimaryKey
    public String _id;
    public String token;
    public String type;
    public String service;
    public String name;
    public String OS;
    public String state;
    public String updatedAt;
    public String createdAt;
    public String active;

}

File: app/src/main/java/org/schulcloud/mobile/models/devices/Device.kt
package org.schulcloud.mobile.models.devices

import com.google.gson.annotations.SerializedName
import io.realm.RealmModel
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

open class Device: RealmObject() {

    @PrimaryKey
    @SerializedName("_id")
    var id: String = ""

    var token: String? = null
    var type: String? = null
    var service: String? = null
    var name: String? = null
    var OS: String? = null
    var state: String? = null
    var updatedAt: String? = null
    var createdAt: String? = null
    var active: String? = null

}